### ISSUE TITLE
Support optional chaining proposal

### ISSUE THREAD

**Issue by lensbart** [gajus/eslint-plugin-jsdoc#182]
Using the [optional chaining proposal](https://github.com/tc39/proposal-optional-chaining) leads `eslint-plugin-jsdoc` to think that the constant being annotated is a function, while it’s not:

```js
/** @const {boolean} test */
const test = something?.find(_ => _)
```

gives the following warnings:

* Missing JSDoc @description declaration.eslint(`jsdoc/require-description`)
* Missing JSDoc @param "_" declaration.eslint(`jsdoc/require-param`)
* Missing JSDoc @returns declaration.eslint(`jsdoc/require-returns`)

while this passes without warnings:

```js
/** @const {boolean} test */
const test = something.find(_ => _)
```

**Comment by golopot** [comment]
This is blocked by #189 . The function `getJSDocComment` that causes the false positive is in eslint core.

**Comment by brettz9** [comment]
#189 is no longer blocking as it has been merged; feel free to take up the issue if you like...
